tests/manager/search/test_external_search.py

```diff
@@ -2040,34 +2033,45 @@ def assert_featured(description, worklist, facets, expect):
         assert data.featured_on_list in last_two
 
         # Up to this point we've been avoiding the random element,
-        # but we can introduce that now by passing in a numeric seed.
-        # In normal usage, the current time is used as the seed.
+        # but we can introduce that now by letting the random_seed
+        # parameter be None.
         #
         # The random element is relatively small, so it mainly acts
         # to rearrange works whose scores were similar before.
         #
-        # The order of the works when using random depends on 4 things:
-        # - The seed
-        # - The id (work_id)
-        # - The index name
-        # - The shard id
-        # If any of those change the order of works in this result may change,
-        # and hence the order of works in this assert must also change
-        # E.g. If the index version changes from v5 to v6, this may affect the order of works queried
-        # Keeping everything else the same, the order of works will remain reproducible across test runs
```

Comment:
Previously this test would fail every time we changed the index name, since the index name is part of the seed for the random function. The changes in this PR make the index name effectively random, since each call to the fixture gets its own index name. So this test was re-written to just test what we need to test here, which is that the random function re-orders the high quality and low quality titles, but doesn't cause them to mix.